assets/js/blocks/cart-checkout/cart-i2/use-view-switcher.tsx

```diff
@@ -23,30 +24,26 @@ export const useViewSwitcher = (
 	const initialView =
 		views?.find( ( view ) => view.default === true ) || views[ 0 ];
 	const [ currentView, setCurrentView ] = useState( initialView );
+	const { selectBlock } = useDispatch( 'core/block-editor' );
 
 	const ViewSwitcherComponent = () => (
-		<BlockControls>
-			<Toolbar>
-				<ToolbarDropdownMenu
-					label={ __(
-						'Switch view',
-						'woo-gutenberg-products-block'
-					) }
-					text={ currentView.label }
-					icon={
-						<Icon
-							srcElement={ eye }
-							style={ { marginRight: '8px' } }
-						/>
-					}
-					controls={ views.map( ( view ) => ( {
-						...view,
-						title: view.label,
-						onClick: () => setCurrentView( view ),
-					} ) ) }
-				/>
-			</Toolbar>
-		</BlockControls>
+		<Toolbar>
+			<ToolbarDropdownMenu
+				label={ __( 'Switch view', 'woo-gutenberg-products-block' ) }
+				text={ currentView.label }
+				icon={
+					<Icon srcElement={ eye } style={ { marginRight: '8px' } } />
+				}
+				controls={ views.map( ( view ) => ( {
+					...view,
+					title: view.label,
+					onClick: () => {
+						setCurrentView( view );
+						selectBlock( clientId );
```

Comment:
Can we not select the parent Cart block and just select the current one? So, filled or empty cart block?

Reply:
How will be know the client ID?

Reply:
`selectBlock( getBlock( clientId ).innerBlocks.find(block => block.name === 'woocommerce/filled-cart-block').clientId )`
But like prettier and not a single line.